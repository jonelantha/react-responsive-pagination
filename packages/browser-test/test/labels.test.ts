import { setupThrowOnError, stringifyWithUndefined } from './helper';

beforeAll(async () => {
  setupThrowOnError(page);

  await page.goto(`${harnessUrl}bootstrap4`);

  await page.setViewportSize({ width: 700, height: 700 });
});

describe('Labels', () => {
  beforeEach(async () => {
    await page.fill('#previousLabelAsJson', 'undefined');

    await page.fill('#nextLabelAsJson', 'undefined');
  });

  test.each([undefined, '<', 'Previous'].map(previousLabel => [previousLabel]))(
    'Setting previousLabel to %p',
    async previousLabel => {
      await page.fill('#previousLabelAsJson', stringifyWithUndefined(previousLabel));

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );

  test.each([undefined, '>', 'Next'].map(nextLabel => [nextLabel]))(
    'Setting nextLabel to %p',
    async nextLabel => {
      await page.fill('#nextLabelAsJson', stringifyWithUndefined(nextLabel));

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );

  test.each([undefined, 'active', ''].map(a11yActiveLabel => [a11yActiveLabel]))(
    'Setting a11yActiveLabel to %p',
    async a11yActiveLabel => {
      await page.fill(
        '#a11yActiveLabelAsJson',
        stringifyWithUndefined(a11yActiveLabel),
      );

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );

  test.each([undefined, true].map(ariaCurrentAttr => [ariaCurrentAttr]))(
    'Setting ariaCurrentAttr to %p',
    async ariaCurrentAttr => {
      await page.fill(
        '#ariaCurrentAttrAsJson',
        stringifyWithUndefined(ariaCurrentAttr),
      );

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );
});
