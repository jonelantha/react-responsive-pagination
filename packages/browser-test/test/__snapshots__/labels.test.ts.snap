// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels Setting a11yActiveLabel to "" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting a11yActiveLabel to "active" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting a11yActiveLabel to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaCurrentAttr to true 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaCurrentAttr to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaNextLabel to "AriaNext" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="AriaNext"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaNextLabel to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaPreviousLabel to "AriaPrevious" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="AriaPrevious"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting ariaPreviousLabel to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting nextLabel to ">" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">&gt;</span></a></li></ul>"`;

exports[`Labels Setting nextLabel to "Next" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#">Next</a></li></ul>"`;

exports[`Labels Setting nextLabel to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting previousLabel to "<" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">&lt;</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting previousLabel to "Previous" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link">Previous</span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Labels Setting previousLabel to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;
